ProcessLib/SmallDeformationNonlocal/SmallDeformationNonlocalFEM.h

```diff
@@ -0,0 +1,870 @@
+        _secondary_data.N.resize(n_integration_points);
+
+        auto const shape_matrices =
+            initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                              IntegrationMethod, DisplacementDim>(
+                e, is_axially_symmetric, _integration_method);
+
+        auto& solid_material =
+            MaterialLib::Solids::selectSolidConstitutiveRelation(
+                _process_data.solid_materials,
+                _process_data.material_ids,
+                e.getID());
+        auto* ehlers_solid_material = dynamic_cast<
+            MaterialLib::Solids::Ehlers::SolidEhlers<DisplacementDim>*>(
+            &solid_material);
+        if (ehlers_solid_material == nullptr)
+        {
+            OGS_FATAL(
+                "The SmallDeformationNonlocalLocal supports only Ehlers "
+                "material at the moment. For other materials the interface "
+                "must be extended first.");
+        }
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            _ip_data.emplace_back(*ehlers_solid_material);
+            auto& ip_data = _ip_data[ip];
+            auto const& sm = shape_matrices[ip];
+            _ip_data[ip].integration_weight =
+                _integration_method.getWeightedPoint(ip).getWeight() *
+                sm.integralMeasure * sm.detJ;
+
+            ip_data.N = sm.N;
+            ip_data.dNdx = sm.dNdx;
+
+            // Initialize current time step values
+            ip_data.sigma.setZero(MathLib::KelvinVector::KelvinVectorDimensions<
+                                  DisplacementDim>::value);
+            ip_data.eps.setZero(MathLib::KelvinVector::KelvinVectorDimensions<
+                                DisplacementDim>::value);
+
+            // Previous time step values are not initialized and are set later.
+            ip_data.sigma_prev.resize(
+                MathLib::KelvinVector::KelvinVectorDimensions<
+                    DisplacementDim>::value);
+            ip_data.eps_prev.resize(
+                MathLib::KelvinVector::KelvinVectorDimensions<
+                    DisplacementDim>::value);
+
+            _secondary_data.N[ip] = shape_matrices[ip].N;
+
+            ip_data.coordinates = getSingleIntegrationPointCoordinates(ip);
+        }
+    }
+
+    std::size_t setIPDataInitialConditions(std::string const& name,
+                                    double const* values,
+                                    int const integration_order) override
+    {
+        if (integration_order !=
+            static_cast<int>(_integration_method.getIntegrationOrder()))
+        {
+            OGS_FATAL(
+                "Setting integration point initial conditions; The integration "
+                "order of the local assembler for element %d is different from "
+                "the integration order in the initial condition.",
+                _element.getID());
+        }
+
+        if (name == "sigma_ip")
```

Comment:
Enum is not an option because each process might call this differently. The function call (and so the comparison) happens once for each element when setting initial conditions.